### PR TITLE
feature(smart-contracts): Card purchaser

### PR DESCRIPTION
**Pull request by julien51**
# Description

Introducing the card purchaser contract.
This contract has a single (stateless) purpose: spend a user's USDC to buy from a lock:
It take 2 signatures:
- one by the user to spend USDC
- one by the user to purchase from a lock
as well as a `callData` from a lock.

### CODE REVIEW

**Comment by julien51** [comment]
Tests coming once I get a first validation about the approach!

File: smart-contracts/contracts/CardPurchaser.sol [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  /**
+   * Constructor
+   * We double the `name` and `version` fields from EIP712 so we can query the contract to get them and OZ's Ownable does not expose them
+   */
+  constructor(
+    address _owner,
+    address _unlockAddress,
+    address _usdc
+  ) EIP712("Card Purchaser", "1") Ownable() {
+    name = "Card Purchaser";
```
Comment (clemsos) [inline smart-contracts/contracts/CardPurchaser.sol:57]: You should be able to access is using `super.name` no?

**Review by clemsos — APPROVED** [review]
LGTM

File: smart-contracts/contracts/CardPurchaser.sol (hunk 1 above)
Comment (julien51) [reply smart-contracts/contracts/CardPurchaser.sol:57]: Let me try but `name` is a `private` var in `EIP712` :(

**Pull request merged**